### PR TITLE
[#215] implement join lobby endpoint according to spec + tests, archi…

### PR DESCRIPTION
**Pull request by TR1NITROTOLUOL**

### CODE REVIEW

File: backend/api/src/test/kotlin/software/shonk/adapters/incoming/ShorkInterpreterControllerV1IT.kt [hunk 1]
```diff
@@ -289,4 +289,56 @@ class ShorkInterpreterControllerV1IT : AbstractControllerTest() {
...
+
+    @Test
+    fun `test join nonexistent lobby`() = runTest {
+        val result =
+            client.post("/api/v1/lobby/0/join") {
+                contentType(ContentType.Application.Json)
+                setBody("{\"playerName\":\"playerA\"}")
+            }
+        assertEquals(HttpStatusCode.Conflict, result.status)
+    }
```
Comment (0815Sailsman) [inline backend/api/src/test/kotlin/software/shonk/adapters/incoming/ShorkInterpreterControllerV1IT.kt:343]: This test is good, but I feel like a 404 would be better here. Opinions? Should we adjust the spec?

**Review by 0815Sailsman — CHANGES_REQUESTED** [review]
Looks mostly good to me, but I would like it, if you added to unit tests on the service level aswell (test the stuff you added in shorkService in the ShorkServiceTest).

File: backend/api/src/test/kotlin/software/shonk/adapters/incoming/ShorkInterpreterControllerV1IT.kt (hunk 1 above)
Comment (flamion) [reply backend/api/src/test/kotlin/software/shonk/adapters/incoming/ShorkInterpreterControllerV1IT.kt:343]: 404 if the lobby does not exist, `conflict` if it's disallowed due to the player already being in the lobby is fitting imo

Comment (0815Sailsman) [reply backend/api/src/test/kotlin/software/shonk/adapters/incoming/ShorkInterpreterControllerV1IT.kt:343]: yes, my thoughts.

I will adjust the docs accordingly then.

**Review by 0815Sailsman — APPROVED** [review]
Nice

**Pull request merged**